src/desktop/LayerSurface.cpp

```diff
@@ -107,6 +107,15 @@ void CLayerSurface::onDestroy() {
         g_pHyprRenderer->damageBox(&geomFixed);
     }
 
+    for (auto& mon : g_pCompositor->m_vRealMonitors) {
+        for (size_t i = 0; i < mon->m_aLayerSurfaceLayers.size(); i++) {
+            auto it = std::find_if(mon->m_aLayerSurfaceLayers[i].begin(), mon->m_aLayerSurfaceLayers[i].end(), [this](auto& ls) { return ls.get() == this; });
```

Comment:
`std::erase_if`

and then you can `return ls.expired() || ls.get() == this`

Reply:
changed